daml/splice-amulet/daml/Splice/AmuletRules.daml

```diff
@@ -1478,32 +1488,37 @@ template TransferPreapproval
          let result = transferResult with meta = None
          pure (TransferPreapproval_SendResult result (Some (Metadata meta)))
 
-    -- Renew the contract as the provider
+    -- Renew the contract as the provider.
+    --
+    -- Renewal durations shorter than `AmuletConfig.transferPreapprovalBaseDuration` do not require
+    -- any CC inputs, as they are implicitly paid for by the traffic fees spent for the renewal transaction.
     choice TransferPreapproval_Renew : TransferPreapproval_RenewResult
       with
         context : PaymentTransferContext
         inputs : [TransferInput]
         newExpiresAt : Time
       controller provider
       do
-         now <- getTime
-         require "Contract is not expired" (expiresAt > now)
-         require "newExpiresAt is after expiresAt" (newExpiresAt > expiresAt)
-         let extension = newExpiresAt `subTime` expiresAt
-         amuletRules <- fetchPublicReferenceData (ForDso dso) context.amuletRules (AmuletRules_Fetch dso)
-         let configUsd = getValueAsOf now amuletRules.configSchedule
-         (amuletPaid, _) <- computeTransferPreapprovalFee extension configUsd context.context dso
-         (transferResult, meta) <- splitAndBurn provider amuletPaid inputs context.context dso "renew TransferPreapproval"
-         transferPreapprovalCid <- create this with
-           lastRenewedAt = now
-           expiresAt = newExpiresAt
-         return TransferPreapproval_RenewResult with meta = Some meta, ..
+        assertWithinDeadline "TransferPreapproval.expiresAt" expiresAt
+        require "newExpiresAt is after expiresAt" (newExpiresAt > expiresAt)
+        (amuletPaid, transferResult, meta) <-
+          payForTransferPreapproval dso provider context inputs expiresAt newExpiresAt
+        transferPreapprovalCid <- create this with
+          -- Last renewed at is only used for tracking purposes, so we assume that the
+          -- extension of the duration happens close enough to the current time so that
+          -- we can use the new expiration time as the last renewed time.
+          -- Thereby avoiding a dependency on `getTime`, which does not work well for external parties.
+          lastRenewedAt = expiresAt
+          expiresAt = newExpiresAt
+        return TransferPreapproval_RenewResult with meta = Some meta, ..
+
+    -- FIXME(QUESTION): should we also provide a simpler renewal choice that does not require a `PaymentTransferContext`?
```

Comment:
hm, does the payment context get in the way? Seems easy enough to construct that I probably wouldn't bother. Just one less thing to support and test